More stepping stones on using arguments and functions: Intro to R, Seeking Help + Data Structures (#787)
swcarpentry/r-novice-gapminder: _episodes_rmd/03-seeking-help.Rmd

help(function_name)
```

For example take a look at the help file for `write.table()`, we will be using a similar function in an upcoming episode. 

```{r}
?write.table()
```

This will load up a help page in RStudio (or as plain text in R by itself).

Each help page is broken down into sections:

 - Description: An extended description of what the function does.

>  - Usage: The arguments of the function and their default values which can be changed.

Reviewer: ```suggestion
 - Usage: The arguments of the function and their default values (which can be changed).
```

 - Arguments: An explanation of the data each argument is expecting.
 - Details: Any important details to be aware of.
 - Value: The data the function returns.
 - See Also: Any related functions you might find useful.
 - Examples: Some examples for how to use the function.

Different functions might have different sections, but these are the main ones you should be aware of. 

> Notice how similar functions call for the same help file: 

Reviewer: ```suggestion
Notice how related functions might call for the same help file: 
```

```{r}
?write.table()
?write.csv()
```

> This is because these functions have very similar applicability and often share the same arguments as inputs to the function.

Reviewer: ```suggestion
This is because these functions have very similar applicability and often share the same arguments as inputs to the function, so package authors often choose to document them together in a single help file.
```

> > used between concatenated terms — by default, a space. The result is a
> > vector as long as the longest argument supplied to `paste`. In contrast,
> > `collapse` specifies that after concatenation the elements are *collapsed*
> > together using the given separator, the result being a single string. 
> >

> > > Its important to call the arguments explicitly by typing out the argument 

Reviewer: ```suggestion
> > It is important to call the arguments explicitly by typing out the argument 
```

> > name e.g `sep = ","` so the function understands to use the "," as a 

> >> seperator and not a term to concatenate.

Reviewer: ```suggestion
> > separator and not a term to concatenate.
```
